torch_np/_funcs.py

```diff
@@ -93,7 +93,20 @@ def fill_diagonal(a, val, wrap=False):
     return _helpers.array_from(result)
 
 
-# ### sorting ###
+def vdot(a, b, /):
+    t_a, t_b = _helpers.to_tensors(a, b)
+    result = _impl.vdot(t_a, t_b)
+    return result.item()
+
+
+def dot(a, b, out=None):
+    t_a, t_b = _helpers.to_tensors(a, b)
+    dtype = _dtypes_impl.result_type_impl((t_a.dtype, t_b.dtype))
+    t_a = t_a.to(dtype)
+    t_b = t_b.to(dtype)
```

Comment:
cast_if_needed

Reply:
Also, why isn't the type cast within `_impl.dot`?

Reply:
Thanks, indeed. Done,  x2